lib/apiservers/portlayer/restapi/handlers/interaction_handlers.go

```diff
@@ -99,8 +99,24 @@ func (i *InteractionHandlersImpl) ContainerResizeHandler(params interaction.Cont
 }
 
 func (i *InteractionHandlersImpl) ContainerSetStdinHandler(params interaction.ContainerSetStdinParams) middleware.Responder {
-	log.Printf("Attempting to get ssh session for container %s stdin", params.ID)
-	sshConn, err := i.attachServer.Get(context.Background(), params.ID, interactionTimeout)
+	var ctxDeadline time.Time
+	var timeout time.Duration
+
+	// Calculate the timeout for the attach if the caller specified a deadline
+	if params.Deadline != nil {
+		ctxDeadline = time.Time(*params.Deadline)
+		timeout = ctxDeadline.Sub(time.Now())
+		log.Printf("Attempting to get ssh session for container %s stdin with deadline %s", params.ID, ctxDeadline.Format(time.UnixDate))
+		if timeout < 0 {
+			e := &models.Error{Message: fmt.Sprintf("Deadline for stdin already passed for container %s", params.ID)}
```

Comment:
It wasn't originally my idea, but I really like it.  If you're trying to force a 2nd party to adhere to your timeouts, a deadline removes all timing uncertainties.